### PR TITLE
libbpf-tools/execsnoop: fix ret type of bpf_probe_read_user*.

### PR DESCRIPTION
**Pull request by opsnull**
The [bpf_probe_read_user_str* manpage](https://man7.org/linux/man-pages/man7/bpf-helpers.7.html) says: 
> On success, the strictly positive length of the output string, including the trailing NUL character. On error, a negative value.

### CODE REVIEW

**Comment by yonghong-song** [comment]
The current code seems okay. The code churn with the same functionality seems not necessary.
The following is an explanation.

For helper like `bpf_probe_read_user*`, return a negative value means failure. But typical all negative error
code is with range of [-4096, -1], so a 'unsigned int' means the error code is a pretty big unsigned int number,
much larger than typical argsize, so comparison like 'unsigned_ret > argsize' means a failure.

Also, if `bpf_probe_read_user*` failed, the dest buffer will be zero out, it can be tested as well in indicate that
it might be a true end, or it might be an error case. In any case, the arg processing can be stopped.

**Comment by opsnull** [comment]
Thanks for the explanation! Although the result is correct, the code is not straightforward to understand. Moreover, these two cases do not seem to conform to general programming practices, such as comparing signed and unsigned types at the same time, not directly checking the return value of the function but relying on internal implementation details.

File: libbpf-tools/execsnoop.bpf.c [hunk 1]
```diff
@@ -73,6 +73,9 @@ int tracepoint__syscalls__sys_enter_execve(struct trace_event_raw_sys_enter* ctx
 	event->args_size = 0;
 
 	ret = bpf_probe_read_user_str(event->args, ARGSIZE, (const char*)ctx->args[0]);
+	if (ret < 0) {
+		return 0
```
Comment (yonghong-song) [inline libbpf-tools/execsnoop.bpf.c:77]: Okay, I changed my mind. Let us try to code based on uapi specification.
Have you compiled the code before submit the patch? There is a missing ';' in the above return statement.

Comment (opsnull) [reply libbpf-tools/execsnoop.bpf.c]: Aha, sloppy, fixed, compile and self-test pass.

**Pull request merged**